sdks/java/core/src/main/java/org/apache/beam/sdk/io/Source.java

```diff
@@ -64,7 +64,7 @@
   /** @deprecated Override {@link #getOutputCoder()} instead. */
   @Deprecated
   public Coder<T> getDefaultOutputCoder() {
-    throw new UnsupportedOperationException("Source needs to override getOutputCoder()");
+    return getOutputCoder();
```

Comment:
This appears to now be mutually recursive with `getOutputCoder()`, which seems inappropriate.